### PR TITLE
3.0: Bug fixes, test rebalancing, pylint fixes

### PR DESCRIPTION
**Pull request by hanwen-pcluste**
See commit descriptions for details

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: cli/.pylintrc [hunk 1]
```diff
@@ -195,7 +197,7 @@ score=yes
 [REFACTORING]
 
 # Maximum number of nested blocks for function / method body
-max-nested-blocks=5
+max-nested-blocks=6
```
Comment (enrico-usai) [inline cli/.pylintrc:200]: I'd avoid extending this, we can instead define a function/method for some of the nested blocks.

File: cli/.pylintrc [hunk 2]
```diff
@@ -158,6 +158,8 @@ disable=print-statement,
         no-self-use,
         no-member,
         unused-import, # handled by flake8
+        no-else-return, # A complete if else block can improve readability even with returns in the body
+        invalid-name, # This would prevent us from using single letter variable
```
Comment (enrico-usai) [inline cli/.pylintrc:162]: I'd keep this check https://www.python.org/dev/peps/pep-0008/#prescriptive-naming-conventions

File: cli/src/pcluster/models/cluster.py [hunk 3]
```diff
@@ -654,4 +654,7 @@ def _add_version_tag(self):
 
     def _get_cfn_tags(self):
         """Return tag list in the format expected by CFN."""
-        return [{"Key": tag.key, "Value": tag.value} for tag in self.config.tags]
+        return [{"Key": tag.key, "Value": tag.value} for tag in self.config.tags] + [
+            {"Key": PCLUSTER_S3_BUCKET_TAG, "Value": self.bucket.name},
+            {"Key": PCLUSTER_S3_CLUSTER_DIR_TAG, "Value": self.bucket.artifact_directory},
```
Comment (enrico-usai) [inline cli/src/pcluster/models/cluster.py:659]: Could you add more description in the commit message? Why do we need this tag?

File: cli/.pylintrc (hunk 1 above)
Comment (hanwen-pcluste) [reply cli/.pylintrc]: Done

File: cli/.pylintrc (hunk 2 above)
Comment (hanwen-pcluste) [reply cli/.pylintrc]: Done

File: cli/src/pcluster/cli_commands/commands.py [hunk 4]
```diff
@@ -310,12 +311,12 @@ def delete(args):  # noqa: C901
         LOGGER.debug("Status: %s", result.stack_status)
         if not args.nowait:
             verified = utils.verify_stack_status(
-                result.stack_name, waiting_states=["DELETE_IN_PROGRESS"], successful_states=["DELETE_COMPLETE"]
+                result.stack_arn, waiting_states=["DELETE_IN_PROGRESS"], successful_states=["DELETE_COMPLETE"]
```
[the remark this answers is not in the document]
Comment (hanwen-pcluste) [reply cli/src/pcluster/cli_commands/commands.py:314]: Using stack_arn, we can describe a stack even if it is deleted. So it is useful for the delete command

File: cli/src/pcluster/models/cluster.py (hunk 3 above)
Comment (hanwen-pcluste) [reply cli/src/pcluster/models/cluster.py:665]: Let's talk about it

**Review by enrico-usai — APPROVED** [review]

**Review by enrico-usai — APPROVED** [review]

**Pull request merged**